pyhf/modifiers/staterror.py

```diff
@@ -50,8 +50,16 @@ def __init__(self,staterr_mods,pdfconfig,mega_mods):
             for mask,inds in zip(staterror_mask, self._staterror_indices):
                 summed_mask = default_backend.sum(mask[:,0,:],axis=0)
                 assert default_backend.shape(summed_mask[summed_mask >  0]) == default_backend.shape(default_backend.astensor(inds))
-                summed_mask[summed_mask >  0] = inds
-                summed_mask[summed_mask == 0] = -1
+                # make masks of > 0 and == 0
+                positive_mask = summed_mask > 0
+                zero_mask = summed_mask == 0
+                # then apply the mask
+                summed_mask[positive_mask] = inds
+                summed_mask[zero_mask] = 0
+                # nb: old code above was
```

Comment:
These **nb** comments (below) probably don't need to be in the code, right? They could just exist in the PR, but we don't need them in the code base.